lib/api/CDataFrameBoostedTreeRunner.cc

```diff
@@ -145,10 +156,19 @@ void CDataFrameBoostedTreeRunner::runImpl(const TStrVec& featureNames,
         return;
     }
 
+    core::CProgramCounters::counter(counter_t::E_DFTPMEstimatedPeakMemoryUsage) =
+        this->estimateMemoryUsage(frame.numberRows(),
+                                  frame.numberRows() / this->numberPartitions(),
+                                  frame.numberColumns());
+
+    core::CStopWatch watch{true};
```

Comment:
Should we not also add a watch tracking outlier detection duration?

Reply:
This is done at a lower level for outlier detection: see for example [this](https://github.com/elastic/ml-cpp/blob/master/lib/maths/COutliers.cc#L895) line.